subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/BuildInitPlugin.java

```diff
@@ -31,61 +32,88 @@
  * The build init plugin.
  */
 public class BuildInitPlugin implements Plugin<Project> {
+
     @Override
-    public void apply(final Project project) {
+    public void apply(Project project) {
         if (project.getParent() == null) {
-            project.getTasks().register("init", InitBuild.class, new Action<InitBuild>() {
-                @Override
-                public void execute(InitBuild initBuild) {
-                    initBuild.setGroup("Build Setup");
-                    initBuild.setDescription("Initializes a new Gradle build.");
-
-                    initBuild.onlyIf(new Spec<Task>() {
-                        @Override
-                        public boolean isSatisfiedBy(Task element) {
-                            Object skippedMsg = reasonToSkip(project);
-                            if (skippedMsg != null) {
-                                project.getLogger().warn((String) skippedMsg);
-                                return false;
-                            }
-
-                            return true;
-                        }
-                    });
-
-                    initBuild.dependsOn(new Callable<String>() {
-                        @Override
-                        public String call() throws Exception {
-                            if (reasonToSkip(project) == null) {
-                                return "wrapper";
-                            } else {
-                                return null;
-                            }
-                        }
-                    });
-                }
+            project.getTasks().register("init", InitBuild.class, initBuild -> {
```

Comment:
We should think about somehow marking this task as 'does not make sense with instant execution'. It doesn't make, well, sense to run this a second time.

Reply:
.. but not in this PR. Maybe we add an issue to the board.